cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -39,11 +39,11 @@ func (c *Controller) syncHandler(key, operation string, object interface{}) erro
 	// Check if the event is for delete and use the spc object that was pushed in the queue
 	// for utilising details from it e.g. delete cas template name for storagepool deletion.
 	if operation == deleteEvent {
-		object := object.(*apis.StoragePoolClaim)
-		if object == nil {
+		spcObject := object.(*apis.StoragePoolClaim)
+		if spcObject == nil {
```

Comment:
Please add a comment that object interface is never nil, hence the nil check is done after typecast only.

Reply:
Sure..

Reply:
@avishnu Little bit of more information.(To correct myself)
The concept is following:
```Interface type of nil is different from nil but all other type of nil has the same type  as that of nil.```
That means, if you pass a nil object of type let us say, `A` to an interface and then try to compare the object for equality with nil,
it will return false.